libs/blocks/brick/brick.js

```diff
@@ -0,0 +1,105 @@
+import { decorateTextOverrides, decorateBlockText, decorateBlockBg, decorateIconStack, decorateButtons } from '../../utils/decorate.js';
+import { createTag } from '../../utils/utils.js';
+
+const blockTypeSizes = {
+  full: ['xxl', 'm', 'l'],
+  default: ['xl', 'm', 'l'],
+};
+const objFitOptions = ['fill', 'contain', 'cover', 'none', 'scale-down'];
+
+function getBlockSize(el) {
+  const sizes = Object.keys(blockTypeSizes);
+  const size = sizes.find((s) => el.classList.contains(`${s}-grid`)) || 'default';
+  return blockTypeSizes[size];
+}
+
+function handleBrickFragment(el) {
+  if (!el.closest('.fragment')) return;
+  const gridVariant = [...el.classList]?.find((c) => c.match(/-grid/));
+  if (!gridVariant || !el.closest('.section.masonry-layout')) return;
+  if (!el.parentNode.classList.contains('.section.masonry-layout')) {
+    el.closest('.section.masonry-layout div:not([class])')?.classList.add(gridVariant);
+  }
+}
+
+function handleSupplementalText(foreground) {
+  if (!foreground.querySelector('.action-area')) return;
+  const nextP = foreground.querySelector('.action-area + p');
+  const lastP = foreground.querySelector('.action-area ~ p:last-child');
+  if (nextP) nextP.className = '';
+  if (lastP) lastP.className = 'supplemental-text';
+}
+
+function setObjectFitAndPos(text, pic, bgEl) {
+  const backgroundConfig = text.split(',').map((c) => c.toLowerCase().trim());
+  const fitOption = objFitOptions.filter((c) => backgroundConfig.includes(c));
+  const focusOption = backgroundConfig.filter((c) => !fitOption.includes(c));
+  if (fitOption) [pic.querySelector('img').style.objectFit] = fitOption;
+  bgEl.innerHTML = '';
+  bgEl.append(pic);
+  bgEl.append(document.createTextNode(focusOption.join(',')));
+}
+
+function handleObjectFit(bgRow) {
+  const bgConfig = bgRow.querySelectorAll('div');
+  [...bgConfig].forEach((r) => {
+    const pic = r.querySelector('picture');
+    if (!pic) return;
+    let text = '';
+    const pchild = r.querySelectorAll('p:not(:empty)');
```

Comment:
Should this be trimmed? [According to MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/:empty) whitespace counts as a child.

Reply:
@rgclayton I have made the suggested change